### Задачки с rosalind.info

Названия директорий соответствуют id задачи.

Для запуска тестового примера набрать:
```shell
run_task.sh -t task_id
```

Если передаем свои данные на вход:
```shell
run_task.sh -t task_id -i <<< "DATA"
cat datafile | run_task.sh -t task_name -i
```

**В некоторых задачах будет принудительно запускаться чтение данных из соответствующего файла в rosalind_data т.к. не удалось обеспечить чтение данных из stdin профильной библиотекой**

###### Флаги:
- `` -t `` id задачи

- `` -l py|r `` выбор языка

- `` -i `` чтение данных из stdin
---

##### Статистика 

| Problem id | Title | Soved |
| :---: | :---: | :---: |
| DNA | Counting DNA Nucleotides | :white_check_mark: |
| RNA | Transcribing DNA into RNA | :white_check_mark: |
| REVC | Complementing a Strand of DNA | :white_check_mark: |
| FIB | Rabbits and Recurrence Relations | :white_check_mark: |
| GC | Computing GC Content | :white_check_mark: |
| HAMM | Counting Point Mutations | :white_check_mark: |
| IPRB | Mendel's First Law | :white_check_mark: |
| CONS | Consensus and Profile	 | :white_check_mark: |
| PROT | Translating RNA into Protein | :white_check_mark: |
| SUBS | Finding a Motif in DNA | :white_check_mark: |
| FIDB | Mortal Fibonacci Rabbit | :white_check_mark: |
| GRPH | Overlap Graphs | :white_check_mark: |
| IEV | Calculating Expected Offspring | :white_check_mark: |
| LCSM | Finding a Shared Motif | :white_check_mark: |
| LIA | Independent Alleles | :white_check_mark: |
| MPRT | Finding a Protein Motif | :x: |
| MRNA | Inferring mRNA from Protein | :white_check_mark: |
| ORF | Open Reading Frames | :x: |
| PERM | Enumerating Gene Orders | :white_check_mark: |
| PRTM | Calculating Protein Mass | :white_check_mark: |
| REVP | Locating Restriction Sites | :white_check_mark: |
| SPLC | RNA Splicing | :white_check_mark: |
| LEXF | Enumerating k-mers Lexicographically | :white_check_mark: |
| LGIS | Longest Increasing Subsequence | :white_check_mark: |
| LONG | Genome Assembly as Shortest Superstring | :white_check_mark: |
| PROB | Introduction to Random Strings | :white_check_mark: |
| KMER | k-Mer Composition | :white_check_mark: |
| TRAN | Transitions and Transversions | :white_check_mark: |
| KMP | Speeding Up Motif Finding | :white_check_mark: |
| PMCH | Perfect Matchings and RNA Secondary Structures | :white_check_mark: |
| SSEQ | Finding a Spliced Motif | :white_check_mark: |
| SPEC | Inferring Protein from Spectrum | :white_check_mark: |
| FULL | Inferring Peptide from Full Spectrum | :white_check_mark: |
| PPER | Partial Permutations | :white_check_mark: |
| LEXV | Ordering Strings of Varying Length Lexicographically | :white_check_mark: |
| SIGN | Enumerating Oriented Gene Orderings | :white_check_mark: |
| SGRA | Using the Spectrum Graph to Infer Peptides | :x: |
| TREE | Completing a Tree | :white_check_mark: |
| INOD | Counting Phylogenetic Ancestors | :white_check_mark: |
| PDST | Creating a Distance Matrix | :white_check_mark: |
| SSET | Counting Subsets | :white_check_mark: |
| ASPC | Introduction to Alternative Splicing | :white_check_mark: |
| SETO | Introduction to Set Operations | :white_check_mark: |
| BDRU | Constructing a De Bruijn Graph | :x: |
