from Bio import SeqIO

datafile = "../rosalind_data/rosalind_splc.txt"

sequences = []
for record in SeqIO.parse(datafile, "fasta"):
    sequences.append(str(record.seq))

genetic_code = {
    'UUU': 'F', 'UUC': 'F', 'UUA': 'L', 'UUG': 'L',
    'CUU': 'L', 'CUC': 'L', 'CUA': 'L', 'CUG': 'L',
    'AUU': 'I', 'AUC': 'I', 'AUA': 'I', 'AUG': 'M',
    'GUU': 'V', 'GUC': 'V', 'GUA': 'V', 'GUG': 'V',
    'UCU': 'S', 'UCC': 'S', 'UCA': 'S', 'UCG': 'S',
    'CCU': 'P', 'CCC': 'P', 'CCA': 'P', 'CCG': 'P',
    'ACU': 'T', 'ACC': 'T', 'ACA': 'T', 'ACG': 'T',
    'GCU': 'A', 'GCC': 'A', 'GCA': 'A', 'GCG': 'A',
    # 'UAU': 'Y', 'UAC': 'Y', 'UAA': '*', 'UAG': '*',
    'CAU': 'H', 'CAC': 'H', 'CAA': 'Q', 'CAG': 'Q',
    'AAU': 'N', 'AAC': 'N', 'AAA': 'K', 'AAG': 'K',
    'GAU': 'D', 'GAC': 'D', 'GAA': 'E', 'GAG': 'E',
    'UGU': 'C', 'UGC': 'C', 'UGA': '*', 'UGG': 'W',
    'CGU': 'R', 'CGC': 'R', 'CGA': 'R', 'CGG': 'R',
    'AGU': 'S', 'AGC': 'S', 'AGA': 'R', 'AGG': 'R',
    'GGU': 'G', 'GGC': 'G', 'GGA': 'G', 'GGG': 'G'
}


def remove_introns(dna_sequence, introns):
    for intron in introns:
        dna_sequence = dna_sequence.replace(intron, '')
    return dna_sequence

def transcribe(dna_sequence):
    # DNA to RNA transcription (T -> U)
    return dna_sequence.replace('T', 'U')

def translate(rna_sequence):
    protein_sequence = ''
    for i in range(0, len(rna_sequence), 3):
        codon = rna_sequence[i:i+3]
        if len(codon) == 3:
            protein_sequence += genetic_code.get(codon, '')
    return protein_sequence


dna_sequence = sequences[0]
introns = sequences[1:]
    
exons = remove_introns(dna_sequence, introns)
    
rna_sequence = transcribe(exons)
protein_sequence = translate(rna_sequence)
    
print(protein_sequence)
